### PR TITLE
Add ability to make robot safe.

### LINKED ISSUE #16: [FEATURE] Make Safe.
Things should be nice and safe.

### PR DESCRIPTION
**Pull request by trickeydan**
Fixes #16

### CODE REVIEW

File: j5/base_robot.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import atexit
+
+from j5.boards import Board
+
+
+class BaseRobot:
+    """A base robot."""
+
+    def make_self(self):
```
Comment (sedders123) [inline j5/base_robot.py]:
```suggestion
    def make_safe(self):
```

File: j5/boards/base.py [hunk 2]
```diff
@@ -23,6 +25,14 @@ def __repr__(self) -> str:
         """A representation of this board."""
         return f"<{self.__class__.__name__} serial={self.serial}>"
 
+    def setup(self):
```
Comment (sedders123) [inline j5/boards/base.py]: Requires caller to remember to call this. Is there a way to make this mandatory? (May not be just a thought)

File: j5/base_robot.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        """Make this robot safe."""
+        BaseRobot.make_all_safe()
+
+    @staticmethod
+    @atexit.register
+    def make_all_safe():
```
Comment (kierdavis) [inline j5/base_robot.py]: Seeing as this method is static, would it make more sense to move this method to the `Board` class?

File: j5/boards/base.py [hunk 4]
```diff
@@ -11,6 +11,8 @@
 class Board(metaclass=ABCMeta):
     """A collection of hardware that has an implementation."""
 
+    BOARDS: List['Board'] = []  # All the boards.
```
Comment (kierdavis) [inline j5/boards/base.py]: Please can we have a comment to explain what the purpose of this list is?

e.g. "This list contains every instance of `Board` that has had its `setup` method called, and is used by `BaseRobot.make_all_safe` when Python exits to ensure every board is made safe."

File: j5/boards/base.py (hunk 2 above)
Comment (trickeydan) [reply j5/boards/base.py]: Not easily, documented in #28 for future fixing as discussed.

File: j5/base_robot.py (hunk 3 above)
Comment (trickeydan) [reply j5/base_robot.py]: I agree.

**Pull request merged**